### PR TITLE
feat(msal-guard): accept angular routes for denied and failed

### LINKED ISSUE #703: MsalGuard to support redirect on failure
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Browser:
- [x] Chrome version 74.0.3729.131 
- [ ] Firefox version XX
- [ ] IE version XX
- [ ] Edge version XX
- [ ] Safari version XX

## Library version

@azure/msal-angular
Library version: 0.1.2

## Expected behavior

It would be neat if the `MsalGuard` would support redirecting to an access denied page. The `CanActivate` interface supports returning a `UrlTree`. I believe this was added in Angular v7.1.

So if we can have `MsalConfig.failureRedirectUri = "access-denied-page"` that would be a lot simpler then having to subscribe to the failure event.

### PR DESCRIPTION
**Pull request by snebjorn**
example config:
```ts
// MsalAngularConfiguration
{
  loginDeniedRoute: '/access-denied',
  loginFailedRoute: '/login-failed',
}
```

fix #703

I was unable to build/run/test as your build guide doesn't work. However a version of this is running in a live system.

### CODE REVIEW

File: lib/msal-angular/src/msal-guard.service.ts [hunk 1]
```diff
@@ -1,103 +1,191 @@
...
+    WindowUtils,
+} from "msal";
 import { MSAL_CONFIG, MSAL_CONFIG_ANGULAR } from "./constants";
+import { MsalAngularConfiguration } from "./msal-angular.configuration";
+import { MsalService } from "./msal.service";
 
-@Injectable()
-export class MsalGuard implements CanActivate {
+@Injectable({
+    providedIn: "root",
```
Comment (snebjorn) [inline lib/msal-angular/src/msal-guard.service.ts:26]: I took the liberty to provide this in root

**Comment by snebjorn** [comment]
Hmm
> Error: Npm failed with return code: 111

Isn't very helpful. Please advice...

**Comment by sameerag** [comment]
cc @jasonnutter

**Comment by jasonnutter** [comment]
The error with the build step is due to the inability for forks to report test coverage to coveralls. Known issue which we will fix soon.

**Comment by snebjorn** [comment]
Aright so not critical, good. Feel free to review at your leisure :)

**Comment by snebjorn** [comment]
@jasonnutter is something on my end blocking the progress of this? Can I do something to help the process along?

**Comment by jasonnutter** [comment]
@snebjorn Apologies for the delay, I've been busy with some other things, but I'll follow up on this soon. I'll let you know if I need anything else, thanks!

**Comment by snebjorn** [comment]
:(

**Comment by FizzBuzz791** [comment]
@jasonnutter - what's happening here? This would be a very useful feature; we'd like to point our multi-tenant users to IT support so that they can make a request to ensure they're in the right tenant, AD group, etc. 

Without this feature / PR it's not straightforward to do.

**Pull request closed without merging**